lib/bio/include/lib/bio.h

```diff
@@ -130,4 +130,18 @@ enum bio_ioctl_num {
     BIO_IOCTL_IS_MAPPED,    /* if supported, returns whether or not the device is memory mapped. */
 };
 
-__END_CDECLS
\ No newline at end of file
+void bio_iter_devices(bool (*callback)(void *, bdev_t *), void *cookie);
```

Comment:
Can you put a comment here that it is called back with the internal bio mutex held, so that callers dont try to do other bio routines inside the callback?

Reply:
Similarly can you add a comment that the bool of the callback tells it to continue calling back or not?

Reply:
Done